### PR TITLE
Add User-Agent customization

### PR DESCRIPTION
**Pull request by cicku**
New field `uag`.

Test:

```
> ./rust-motd  motd.toml
fl=x
h=cloudflare.com
ip=x
ts=x
visit_scheme=https
uag=uag testing
colo=IAD
sliver=none
http=http/1.1
loc=US
tls=TLSv1.3
sni=plaintext
warp=off
gateway=off
rbi=off
kex=X25519

Last updated: 2023-03-01 x
```

### CODE REVIEW

File: src/components/weather.rs [hunk 1]
```diff
@@ -77,9 +77,15 @@ impl Weather {
             }
             None => ureq::AgentBuilder::new().build(),
         };
+
+        let uag = match self.uag {
+            Some(uag) => uag,
+            None => {let uag = String::from("curl"); uag}
```
Comment (MarcelRobitaille) [inline src/components/weather.rs:83]: `let uag = String::from("curl"); uag` this seems redundant. Wouldn't just `None => String::from("curl")` work?

**Review by MarcelRobitaille — CHANGES_REQUESTED** [review]
Thanks for the pull request. I would prefer something more descriptive than `uag`. Please rename the config option and variable to `user_agent` or `custom_user_agent`.

**Comment by MarcelRobitaille** [comment]
I think you are also missing the new key in [the Weather configuration struct](https://github.com/rust-motd/rust-motd/blob/main/src/components/weather.rs#L12-L21).

**Comment by cicku** [comment]
I forgot to commit all changes🤦‍♂️

Fixed.

File: example_config.toml [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 # [weather]
 # url = "https://wttr.in/New+York,New+York?0"
+# uag = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"
```
Comment (MarcelRobitaille) [inline example_config.toml:17]: Please fix uag -> user_agent in the example config as well.

**Review by MarcelRobitaille — CHANGES_REQUESTED** [review]

**Comment by MarcelRobitaille** [comment]
Also, please document the new option in the README [here](https://github.com/rust-motd/rust-motd/blob/main/README.md?plain=1#L45).

File: example_config.toml (hunk 2 above)
Comment (cicku) [reply example_config.toml]: Done

**Pull request merged**